_source/_posts/2025-01-23-terraform-powershell-lab.md

```diff
@@ -0,0 +1,362 @@
+
+2. Create an app integration for the Terraform lab.
+
+    a. Select Create App Integration.
+    
+    b. Select API Services.
+    
+    c. Select Next.
+    
+    d. Enter the application name "Terraform lab."
+    
+    e. Select Save.
+    
+3. Generate the client credentials.
+    
+    a. In the Client Credentials box, select Edit.
+    
+    b. Under Client authentication, select "Public key / Private key."
+    
+    c. Select Add Key.
+    
+    d. Select Generate new key.
+    
+    e. Under "Private Key - Copy this!" select PEM. 
+    **Note**: the PEM key begins with the line `-----BEGIN PRIVATE KEY-----`. Make sure you're looking at the PEM, not the JSON. 
+    
+    f. Select Copy to clipboard.
+
+4. On your computer, paste the key into a file and save it as `key.pem`
+
+   **Important:** The key is saved locally to simplify the lab. You should save the key to an appropriate secrets management solution when working with production environments. If you do not have secrets management, [this blog post](/blog/2024/10/11/terraform-ci-cd) shares one way to set it up.
+
+5. Complete the configuration steps.
+    
+    a. From the Okta Admin console, select Done.
+    
+    b. Select Save.
+    
+    c. In the "Existing client secrets will no longer be used" popup, select Save.
+
+6. From the Okta API Scopes tab, grant the following scopes:
+    - `okta.appGrants.manage`
+    - `okta.apps.manage`
+    - `okta.oauthIntegrations.manage`
+    - `okta.policies.read`
+    - `okta.users.manage`
+
+7. Modify the admin roles for the Terraform lab app.
+    
+    a. Go to the Admin Roles tab.
+    
+    b. Select Edit Assignments.
+    
+    c. Select the role of Super Adminstrator from the dropdown menu.
+    
+    d. Select Save Changes.
+
+8. In your VM, create the file `main.tf`:
+
+   ```hcl
+   terraform {
+     required_providers {
+       okta = {
+         source = "okta/okta"
+       }
+     }
+   }
+
+   variable "org_id" {
+     default = "ORGID"
+   }
+
+   provider "okta" {
+     org_name  = var.org_id
+     base_url  = "oktapreview.com"
+     client_id = "CLIENTID"
+     scopes = ["okta.apps.manage", "okta.appGrants.manage",
+       "okta.oauthIntegrations.manage", "okta.users.manage",
+     "okta.policies.read"]
+     private_key = file("key.pem")
+   }
+
+   ```
+9. In your `main.tf` file, replace `ORGID` with the subdomain for your Okta org.
+
+   **Note:** The subdomain is between `https://` and `.oktapreview.com`. In the example below, the subdomain is `oktaice0000000`.
+
+10. In your `main.tf` file, replace `CLIENTID` with the client ID of your Terraform lab app.
+
+    **Note:** Go to Applications > Applications to find the client ID of your Terraform lab app.
+
+11. Save the `main.tf` file.
+12. Initiate and run the terraform plan.
+    
+    a. In your terminal, run the following commands:
+      
+      ```
+      terraform init
+      terraform plan
+      ```
```

Comment:
I think you're missing a newline or something. the syntax highlighting in GH is catching a problem here